Outing ratings

asaunier opened the pull request:
Related to https://github.com/c2corg/v6_ui/issues/1800
and also to the default outings values issues https://github.com/c2corg/v6_ui/issues/499 and https://github.com/c2corg/v6_ui/issues/1797

API PR at https://github.com/c2corg/v6_api/pull/696

Still todo:
- [x] add ratings filters in the outing advanced search page
- [x] add other default values from the first associated route

asaunier commented:
TODO: add other default values from the first associated route

asaunier commented:
@julsbreakdown The PR is ready to be reviewed. What it does:
* add rating fields in the outing editing form.
* default rating values are retrieved from the first associated route.
* by the way I have added some other default values (elevation min/max, height diff up/down) + in the route editing form: default values for elevation min/max depending on the type of waypoint.
* add rating filters to the outing advanced search page.

@stef74 @desnoes Concerning the last point, I am not fully satisfied by the position of the new rating filters but it's hard to have something consistent and nice when the number of filters varies depending on the selected activities.
I have the feeling the filters section is getting a bit messy: perhaps we should improve it to group filters in foldable sections? Perhaps some filters could be moved or removed?
Please note that activity-specific filters are not reset when updating selected activities. It's already the case in prod with the currently available activity-specific filters (eg. the ratings in the route advanced search page).

stef74 commented:
@asaunier Agree with you ... filter section is very "unreadable". The simple solution si probably to create section foldable. On mobile is very difficult to use. 
We can start by a simple section "other" ...

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -56,29 +59,26 @@
-        <app-slider filter="oparka" filters-list="filtersCtrl.filters" max="5000"></app-slider>
+      <div class="col-xs-12 col-sm-6 filter"
+           ng-if="filtersCtrl.filters.act.length
+               && (filtersCtrl.filters.act.indexOf('snow_ice_mixed') !== -1 ||
```
shouldn't && be on the previous line?

asaunier replied:
It could. But I have reused the way it is done for instance in https://github.com/c2corg/v6_ui/blob/d5ef793673d790adffcea735525ce304f504b971/c2corg_ui/templates/route/filters.html
I am not sure it is worth making such changes everywhere :P

julsbreakdown replied:
I don't really mind, but i've been told that it is better that way =)

asaunier replied:
I think it is better to be consistent and use the same notation than in the other files. Or we could change all files but I don't think it is worth it.

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -87,12 +87,104 @@
-          <span></span>
-        </a>
+      <div ng-if="filtersCtrl.filters.act.length
+               && (filtersCtrl.filters.act.indexOf('snow_ice_mixed') !== -1 ||
```
same here

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -87,12 +87,104 @@
+      </div>
+
+      <div ng-if="filtersCtrl.filters.act.length
+               && filtersCtrl.filters.act.indexOf('rock_climbing') !== -1">
```
same here

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -87,12 +87,104 @@
+      </div>
+
+      <div ng-if="filtersCtrl.filters.act.length
+               && filtersCtrl.filters.act.indexOf('skitouring') !== -1">
```
same here

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -87,12 +87,104 @@
+      </div>
+
+      <div ng-if="filtersCtrl.filters.act.length
+               && filtersCtrl.filters.act.indexOf('ice_climbing') !== -1">
```
same here

julsbreakdown replied:
check them all =)

julsbreakdown commented on requirements.txt:
```diff
@@ -27,6 +27,6 @@ gaiohttp-websocket==0.1.1
 # for development use a local checkout
 # -e ../v6_common
-git+https://github.com/c2corg/v6_common.git@b8eca73
+git+https://github.com/c2corg/v6_common.git@b9a11e3
```
should now be ca2a1be

asaunier replied:
Right! I will make the change. Don't forget to do it in your API PR as well.

julsbreakdown commented on c2corg_ui/templates/outing/edit.html:
```diff
@@ -1,7 +1,10 @@
+    route_duration_types, access_conditions, glacier_ratings, \
+    via_ferrata_ratings, mtb_up_ratings, mtb_down_ratings, ski_ratings, \
+    global_ratings, engagement_ratings, ice_ratings, equipment_ratings, \
+    hiking_ratings, snowshoe_ratings, climbing_ratings
```
Why add climbing_ratings?

julsbreakdown commented on c2corg_ui/templates/outing/edit.html:
```diff
@@ -230,11 +234,161 @@ <h5 translate ng-show="outing.associations.routes.length == 0">Choose at least o
+                 ng-if="outing.activities.indexOf('rock_climbing') > -1">
+              <label app-context-help context-help-title="{{'rock_free_rating' | translate}}" context-help-content-url="/static/partials/contexthelp/route-rock_free_rating.html"><span translate>rock_free_rating</span></label>
+              <div class="input-group">
+                <select class="form-control" ng-model="outing.rock_free_rating" ng-options="rat as rat for rat in ${climbing_ratings}"><option value=""></option></select>
```
Maybe i am wrong but i dont see way climb_ratings is used

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -1,6 +1,9 @@
-    condition_ratings, glacier_ratings, avalanche_signs
+from c2corg_common.attributes import default_langs, quality_types, activities, \
+    frequentation_types, condition_ratings, glacier_ratings, avalanche_signs, \
+    climbing_ratings, ski_ratings, global_ratings, engagement_ratings, equipment_ratings, \
```
Again I dont see why climbing_ratings is used

asaunier replied:
It's used at https://github.com/c2corg/v6_ui/blob/d5ef793673d790adffcea735525ce304f504b971/c2corg_ui/templates/outing/filters.html#L123 for ``rock_free_rating``.
Several ratings use the same climbing_ratings values defined at https://github.com/c2corg/v6_common/blob/master/c2corg_common/attributes.py#L224

julsbreakdown replied:
roger that

julsbreakdown commented on c2corg_ui/templates/outing/filters.html:
```diff
@@ -87,12 +87,104 @@
+        <div class="col-xs-12 col-sm-6 filter">
+          <label translate>rock_free_rating</label><br>
+          <app-slider filter="frat" filters-list="filtersCtrl.filters" unit=""
+            values-list="['${"','".join(climbing_ratings) |n}']"></app-slider>
```
rock_free_rating?

asaunier commented:
I have pushed 2 new fixup commits caf7af8 and 71d2959 because I have realized that some of the previous changes had introduced bugs (eg. it was no longer possible to associate a second WP to a route) and to make sure the ratings default setting was done only when creating the document, not when updating: before the last changes, if one edited an outing, removed all associated routes and associated a route again, the default ratings from the last route were unexpectedly applied, overridding existing values.

asaunier commented:
@julsbreakdown The PR is ready to be reviewed.

asaunier commented:
> The only minor ? aspect, is that when we open an outing we do not see the rating directly in the outing page.

Indeed. But displaying those new attributes either in cards or in the document page itself has not been ordered and will be done by the association. Especially because it must be integrated in the new layout https://github.com/c2corg/v6_ui/pull/1783 (most likely in a future PR).

julsbreakdown commented:
Ok well go merge then =)

asaunier commented:
Thanks, I'll merge when #1783 will be merged and released.

asaunier commented:
I have rebased the branch and fixed a conflict caused by the recent merging of #1783 
I prefer to wait a little before merging the PR, in case additional fixes are needed before the new layout goes to prod (see https://github.com/c2corg/v6_ui/issues/1808).
